ADAL/tests/integration/legacytest/ADLegacyKeychainTokenCache.m

```diff
@@ -189,7 +189,7 @@ - (BOOL)saveWipeTokenData:(ADAuthenticationError * __nullable __autoreleasing *
                                 @"wipeTime" : [NSDate date]
                                 };
 
-    NSData *wipeData = [NSKeyedArchiver archivedDataWithRootObject:wipeInfo];
+    NSData *wipeData = [NSKeyedArchiver archivedDataWithRootObject:wipeInfo requiringSecureCoding:YES error:nil];
```

Comment:
this is test code only. Do we want to keep it as is for backward compatibility?

Reply:
@oldalton nice observation, I made the change considering iOS 12 as the deployment target. In iOS12+ the method [archiveDataWithRootObject:](https://developer.apple.com/documentation/foundation/nskeyedarchiver/1413189-archiveddatawithrootobject) is deprecated.
According to [archivedDataWithRootObject:requiringSecureCoding:error:](https://developer.apple.com/documentation/foundation/nskeyedarchiver/2962880-archiveddatawithrootobject)is still available in iOS 11+, hence I made the replacement. The discussion in the docs suggest : 
![image](https://user-images.githubusercontent.com/6675470/91919834-3f081d00-ec95-11ea-85c0-2c8af55858ba.png)

Do you think I should make these changes when deployment target is set to iOS12?

Reply:
I think we should make this change in production code (all code outside of tests/ directory). This particular class is a test code class explicitly used to test backward compatibility with very old ADAL library. So this one we can keep as is and silence the warnings since the goal of existence of this class is to test compatibility between old ADAL library and new caching logic that lives in common core.

Reply:
I see, I have reverted changes to legacy test cases in the next commit (cc40a79). Also removed deprecation warning suppression.